File: Python/0344. Reverse String.py
#不要直接用reverse

class Solution(object):
    def reverseString(self, s):
        left, right = 0, len(s) - 1

        #用while比for complexity低

        while left < right:
            s[left], s[right] = s[right], s[left]
            left += 1
            right -= 1

File: Java/4. String/0344. Reverse String.java
// similar - 206: reverse linked list, 2 pts, easy

class Solution {
    public void reverseString(char[] s) {
        int left = 0;
        int right = s.length - 1;
        while (left < right) {
            char temp = s[left];
            s[left] = s[right];
            s[right] = temp;
            left++;
            right--;
        }

    }

    // 2nd 240627
    public void reverseString(char[] s) {
        int left = 0;
        int right = s.length - 1;
        while (left < right) {
            char temp = s[left]; // use char!
            s[left] = s[right];
            s[right] = temp;
            left++;
            right--;
        }
    }

}